src/ipz_parser.cpp

```diff
@@ -440,4 +440,80 @@ types::VPDMapVariant IpzVpdParser::parse()
     }
 }
 
+types::BinaryVector
+    IpzVpdParser::getKeywordValueFromRecord(const types::Record& i_record,
+                                            const types::Keyword& i_keyword,
+                                            auto i_recordDataOffset)
+{
+    (void)i_record;
+    (void)i_keyword;
+    (void)i_recordDataOffset;
+
+    return types::BinaryVector();
+}
+
+types::RecordData
+    IpzVpdParser::getRecordDataFromPT(const types::Record& i_record,
+                                      auto i_vtocOffset)
+{
+    (void)i_record;
+    (void)i_vtocOffset;
+
+    return types::RecordData();
+}
+
+types::DbusVariantType IpzVpdParser::readKeywordFromHardware(
+    const types::ReadVpdParams i_paramsToReadData)
+{
+    // Extract record and keyword from i_paramsToReadData
+    types::Record l_record;
+    types::Keyword l_keyword;
+
+    if (const types::IpzType* l_ipzData =
+            std::get_if<types::IpzType>(&i_paramsToReadData))
+    {
+        l_record = std::get<0>(*l_ipzData);
+        l_keyword = std::get<1>(*l_ipzData);
+    }
+    else
+    {
+        logging::logMessage("Given VPD is not of type IPZ. Abort read.");
+        throw types::DeviceError::ReadFailure();
```

Comment:
Why it is a device error if data set by the user is wrong? I would suggest a generic runtime error or invalid argument would suit more.